From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Fri, 4 Nov 2022 14:23:05 -0400
Subject: [PATCH] qtgui: Remove all traces of Python from include/lib.

After commit 956f332 removing pyqtwidget in favor of only qtwidget,
Python has not actually been needed to compile gr-qtgui (other than the
Python bindings). This removes all remaining traces of Python, including
including the Python.h header and linking with Python.

Signed-off-by: Ryan Volz <ryan.volz@gmail.com>
---
 gr-qtgui/include/gnuradio/qtgui/ber_sink_b.h         |  4 ----
 gr-qtgui/include/gnuradio/qtgui/const_sink_c.h       |  4 ----
 gr-qtgui/include/gnuradio/qtgui/edit_box_msg.h       |  4 ----
 gr-qtgui/include/gnuradio/qtgui/eye_sink_c.h         |  4 ----
 gr-qtgui/include/gnuradio/qtgui/eye_sink_f.h         |  4 ----
 gr-qtgui/include/gnuradio/qtgui/freq_sink_c.h        |  7 -------
 gr-qtgui/include/gnuradio/qtgui/freq_sink_f.h        |  7 -------
 gr-qtgui/include/gnuradio/qtgui/histogram_sink_f.h   |  4 ----
 gr-qtgui/include/gnuradio/qtgui/number_sink.h        |  7 -------
 gr-qtgui/include/gnuradio/qtgui/sink_c.h             |  7 -------
 gr-qtgui/include/gnuradio/qtgui/sink_f.h             |  7 -------
 gr-qtgui/include/gnuradio/qtgui/time_raster_sink_b.h |  4 ----
 gr-qtgui/include/gnuradio/qtgui/time_raster_sink_f.h |  4 ----
 gr-qtgui/include/gnuradio/qtgui/time_sink_c.h        |  4 ----
 gr-qtgui/include/gnuradio/qtgui/time_sink_f.h        |  4 ----
 gr-qtgui/include/gnuradio/qtgui/vector_sink_f.h      |  4 ----
 gr-qtgui/include/gnuradio/qtgui/waterfall_sink_c.h   |  7 -------
 gr-qtgui/include/gnuradio/qtgui/waterfall_sink_f.h   |  7 -------
 gr-qtgui/lib/CMakeLists.txt                          | 12 ------------
 gr-qtgui/python/qtgui/bindings/sink_c_python.cc      |  2 +-
 20 files changed, 1 insertion(+), 106 deletions(-)

diff --git a/gr-qtgui/include/gnuradio/qtgui/ber_sink_b.h b/gr-qtgui/include/gnuradio/qtgui/ber_sink_b.h
index 8f44c56af..d076ce09e 100644
--- a/gr-qtgui/include/gnuradio/qtgui/ber_sink_b.h
+++ b/gr-qtgui/include/gnuradio/qtgui/ber_sink_b.h
@@ -11,10 +11,6 @@
 #ifndef INCLUDED_QTGUI_BER_SINK_B_H
 #define INCLUDED_QTGUI_BER_SINK_B_H
 
-#ifdef ENABLE_PYTHON
-#include <Python.h>
-#endif
-
 #include <gnuradio/block.h>
 #include <gnuradio/qtgui/api.h>
 #include <qapplication.h>
diff --git a/gr-qtgui/include/gnuradio/qtgui/const_sink_c.h b/gr-qtgui/include/gnuradio/qtgui/const_sink_c.h
index a1dd5a452..21cd56736 100644
--- a/gr-qtgui/include/gnuradio/qtgui/const_sink_c.h
+++ b/gr-qtgui/include/gnuradio/qtgui/const_sink_c.h
@@ -11,10 +11,6 @@
 #ifndef INCLUDED_QTGUI_CONST_SINK_C_H
 #define INCLUDED_QTGUI_CONST_SINK_C_H
 
-#ifdef ENABLE_PYTHON
-#include <Python.h>
-#endif
-
 #include <gnuradio/filter/firdes.h>
 #include <gnuradio/qtgui/api.h>
 #include <gnuradio/qtgui/trigger_mode.h>
diff --git a/gr-qtgui/include/gnuradio/qtgui/edit_box_msg.h b/gr-qtgui/include/gnuradio/qtgui/edit_box_msg.h
index e3fe7485d..9a6c92686 100644
--- a/gr-qtgui/include/gnuradio/qtgui/edit_box_msg.h
+++ b/gr-qtgui/include/gnuradio/qtgui/edit_box_msg.h
@@ -11,10 +11,6 @@
 #ifndef INCLUDED_QTGUI_EDIT_BOX_MSG_H
 #define INCLUDED_QTGUI_EDIT_BOX_MSG_H
 
-#ifdef ENABLE_PYTHON
-#include <Python.h>
-#endif
-
 #include <gnuradio/block.h>
 #include <gnuradio/qtgui/api.h>
 #include <gnuradio/qtgui/qtgui_types.h>
diff --git a/gr-qtgui/include/gnuradio/qtgui/eye_sink_c.h b/gr-qtgui/include/gnuradio/qtgui/eye_sink_c.h
index a1ff4f60d..2e4507b40 100644
--- a/gr-qtgui/include/gnuradio/qtgui/eye_sink_c.h
+++ b/gr-qtgui/include/gnuradio/qtgui/eye_sink_c.h
@@ -10,10 +10,6 @@
 #ifndef INCLUDED_QTGUI_EYE_SINK_C_H
 #define INCLUDED_QTGUI_EYE_SINK_C_H
 
-#ifdef ENABLE_PYTHON
-#include <Python.h>
-#endif
-
 #include <gnuradio/qtgui/api.h>
 #include <gnuradio/qtgui/trigger_mode.h>
 #include <gnuradio/sync_block.h>
diff --git a/gr-qtgui/include/gnuradio/qtgui/eye_sink_f.h b/gr-qtgui/include/gnuradio/qtgui/eye_sink_f.h
index b13e06d7c..21fcd60b8 100644
--- a/gr-qtgui/include/gnuradio/qtgui/eye_sink_f.h
+++ b/gr-qtgui/include/gnuradio/qtgui/eye_sink_f.h
@@ -10,10 +10,6 @@
 #ifndef INCLUDED_QTGUI_EYE_SINK_F_H
 #define INCLUDED_QTGUI_EYE_SINK_F_H
 
-#ifdef ENABLE_PYTHON
-#include <Python.h>
-#endif
-
 #include <gnuradio/qtgui/api.h>
 #include <gnuradio/qtgui/trigger_mode.h>
 #include <gnuradio/sync_block.h>
diff --git a/gr-qtgui/include/gnuradio/qtgui/freq_sink_c.h b/gr-qtgui/include/gnuradio/qtgui/freq_sink_c.h
index 1755004f5..7f81e5d7d 100644
--- a/gr-qtgui/include/gnuradio/qtgui/freq_sink_c.h
+++ b/gr-qtgui/include/gnuradio/qtgui/freq_sink_c.h
@@ -11,13 +11,6 @@
 #ifndef INCLUDED_QTGUI_FREQ_SINK_C_H
 #define INCLUDED_QTGUI_FREQ_SINK_C_H
 
-#ifdef ENABLE_PYTHON
-#pragma push_macro("slots")
-#undef slots
-#include "Python.h"
-#pragma pop_macro("slots")
-#endif
-
 #include <gnuradio/fft/window.h>
 #include <gnuradio/qtgui/api.h>
 #include <gnuradio/qtgui/trigger_mode.h>
diff --git a/gr-qtgui/include/gnuradio/qtgui/freq_sink_f.h b/gr-qtgui/include/gnuradio/qtgui/freq_sink_f.h
index fb3143126..725b4bce2 100644
--- a/gr-qtgui/include/gnuradio/qtgui/freq_sink_f.h
+++ b/gr-qtgui/include/gnuradio/qtgui/freq_sink_f.h
@@ -11,13 +11,6 @@
 #ifndef INCLUDED_QTGUI_FREQ_SINK_F_H
 #define INCLUDED_QTGUI_FREQ_SINK_F_H
 
-#ifdef ENABLE_PYTHON
-#pragma push_macro("slots")
-#undef slots
-#include "Python.h"
-#pragma pop_macro("slots")
-#endif
-
 #include <gnuradio/fft/window.h>
 #include <gnuradio/qtgui/api.h>
 #include <gnuradio/qtgui/trigger_mode.h>
diff --git a/gr-qtgui/include/gnuradio/qtgui/histogram_sink_f.h b/gr-qtgui/include/gnuradio/qtgui/histogram_sink_f.h
index 00fa1f808..82bee73fe 100644
--- a/gr-qtgui/include/gnuradio/qtgui/histogram_sink_f.h
+++ b/gr-qtgui/include/gnuradio/qtgui/histogram_sink_f.h
@@ -11,10 +11,6 @@
 #ifndef INCLUDED_QTGUI_HISTOGRAM_SINK_F_H
 #define INCLUDED_QTGUI_HISTOGRAM_SINK_F_H
 
-#ifdef ENABLE_PYTHON
-#include <Python.h>
-#endif
-
 #include <gnuradio/qtgui/api.h>
 #include <gnuradio/sync_block.h>
 #include <qapplication.h>
diff --git a/gr-qtgui/include/gnuradio/qtgui/number_sink.h b/gr-qtgui/include/gnuradio/qtgui/number_sink.h
index 57f534d80..64d8d600e 100644
--- a/gr-qtgui/include/gnuradio/qtgui/number_sink.h
+++ b/gr-qtgui/include/gnuradio/qtgui/number_sink.h
@@ -11,13 +11,6 @@
 #ifndef INCLUDED_QTGUI_NUMBER_SINK_H
 #define INCLUDED_QTGUI_NUMBER_SINK_H
 
-#ifdef ENABLE_PYTHON
-#pragma push_macro("slots")
-#undef slots
-#include "Python.h"
-#pragma pop_macro("slots")
-#endif
-
 #include <gnuradio/qtgui/api.h>
 #include <gnuradio/qtgui/qtgui_types.h>
 #include <gnuradio/qtgui/trigger_mode.h>
diff --git a/gr-qtgui/include/gnuradio/qtgui/sink_c.h b/gr-qtgui/include/gnuradio/qtgui/sink_c.h
index e0f815e72..b4729ab2a 100644
--- a/gr-qtgui/include/gnuradio/qtgui/sink_c.h
+++ b/gr-qtgui/include/gnuradio/qtgui/sink_c.h
@@ -11,13 +11,6 @@
 #ifndef INCLUDED_QTGUI_SINK_C_H
 #define INCLUDED_QTGUI_SINK_C_H
 
-#ifdef ENABLE_PYTHON
-#pragma push_macro("slots")
-#undef slots
-#include "Python.h"
-#pragma pop_macro("slots")
-#endif
-
 #include <gnuradio/block.h>
 #include <gnuradio/qtgui/api.h>
 #include <qapplication.h>
diff --git a/gr-qtgui/include/gnuradio/qtgui/sink_f.h b/gr-qtgui/include/gnuradio/qtgui/sink_f.h
index add673ee1..1bb6030c3 100644
--- a/gr-qtgui/include/gnuradio/qtgui/sink_f.h
+++ b/gr-qtgui/include/gnuradio/qtgui/sink_f.h
@@ -11,13 +11,6 @@
 #ifndef INCLUDED_QTGUI_SINK_F_H
 #define INCLUDED_QTGUI_SINK_F_H
 
-#ifdef ENABLE_PYTHON
-#pragma push_macro("slots")
-#undef slots
-#include "Python.h"
-#pragma pop_macro("slots")
-#endif
-
 #include <gnuradio/block.h>
 #include <gnuradio/qtgui/api.h>
 #include <qapplication.h>
diff --git a/gr-qtgui/include/gnuradio/qtgui/time_raster_sink_b.h b/gr-qtgui/include/gnuradio/qtgui/time_raster_sink_b.h
index f6b885e2c..b655949d1 100644
--- a/gr-qtgui/include/gnuradio/qtgui/time_raster_sink_b.h
+++ b/gr-qtgui/include/gnuradio/qtgui/time_raster_sink_b.h
@@ -11,10 +11,6 @@
 #ifndef INCLUDED_QTGUI_TIME_RASTER_SINK_B_H
 #define INCLUDED_QTGUI_TIME_RASTER_SINK_B_H
 
-#ifdef ENABLE_PYTHON
-#include <Python.h>
-#endif
-
 #include <gnuradio/qtgui/api.h>
 #include <gnuradio/sync_block.h>
 #include <qapplication.h>
diff --git a/gr-qtgui/include/gnuradio/qtgui/time_raster_sink_f.h b/gr-qtgui/include/gnuradio/qtgui/time_raster_sink_f.h
index 45b28e3ca..352618802 100644
--- a/gr-qtgui/include/gnuradio/qtgui/time_raster_sink_f.h
+++ b/gr-qtgui/include/gnuradio/qtgui/time_raster_sink_f.h
@@ -11,10 +11,6 @@
 #ifndef INCLUDED_QTGUI_TIME_RASTER_SINK_F_H
 #define INCLUDED_QTGUI_TIME_RASTER_SINK_F_H
 
-#ifdef ENABLE_PYTHON
-#include <Python.h>
-#endif
-
 #include <gnuradio/qtgui/api.h>
 #include <gnuradio/sync_block.h>
 #include <qapplication.h>
diff --git a/gr-qtgui/include/gnuradio/qtgui/time_sink_c.h b/gr-qtgui/include/gnuradio/qtgui/time_sink_c.h
index a8507e0b6..58be28dfe 100644
--- a/gr-qtgui/include/gnuradio/qtgui/time_sink_c.h
+++ b/gr-qtgui/include/gnuradio/qtgui/time_sink_c.h
@@ -11,10 +11,6 @@
 #ifndef INCLUDED_QTGUI_TIME_SINK_C_H
 #define INCLUDED_QTGUI_TIME_SINK_C_H
 
-#ifdef ENABLE_PYTHON
-#include <Python.h>
-#endif
-
 #include <gnuradio/qtgui/api.h>
 #include <gnuradio/qtgui/trigger_mode.h>
 #include <gnuradio/sync_block.h>
diff --git a/gr-qtgui/include/gnuradio/qtgui/time_sink_f.h b/gr-qtgui/include/gnuradio/qtgui/time_sink_f.h
index cb00348a4..dc8c0a402 100644
--- a/gr-qtgui/include/gnuradio/qtgui/time_sink_f.h
+++ b/gr-qtgui/include/gnuradio/qtgui/time_sink_f.h
@@ -11,10 +11,6 @@
 #ifndef INCLUDED_QTGUI_TIME_SINK_F_H
 #define INCLUDED_QTGUI_TIME_SINK_F_H
 
-#ifdef ENABLE_PYTHON
-#include <Python.h>
-#endif
-
 #include <gnuradio/qtgui/api.h>
 #include <gnuradio/qtgui/trigger_mode.h>
 #include <gnuradio/sync_block.h>
diff --git a/gr-qtgui/include/gnuradio/qtgui/vector_sink_f.h b/gr-qtgui/include/gnuradio/qtgui/vector_sink_f.h
index fb1683436..cab1713e9 100644
--- a/gr-qtgui/include/gnuradio/qtgui/vector_sink_f.h
+++ b/gr-qtgui/include/gnuradio/qtgui/vector_sink_f.h
@@ -11,10 +11,6 @@
 #ifndef INCLUDED_QTGUI_VECTOR_SINK_F_H
 #define INCLUDED_QTGUI_VECTOR_SINK_F_H
 
-#ifdef ENABLE_PYTHON
-#include <Python.h>
-#endif
-
 #include <gnuradio/qtgui/api.h>
 #include <gnuradio/sync_block.h>
 #include <qapplication.h>
diff --git a/gr-qtgui/include/gnuradio/qtgui/waterfall_sink_c.h b/gr-qtgui/include/gnuradio/qtgui/waterfall_sink_c.h
index 7c683fb95..f5870738e 100644
--- a/gr-qtgui/include/gnuradio/qtgui/waterfall_sink_c.h
+++ b/gr-qtgui/include/gnuradio/qtgui/waterfall_sink_c.h
@@ -11,13 +11,6 @@
 #ifndef INCLUDED_QTGUI_WATERFALL_SINK_C_H
 #define INCLUDED_QTGUI_WATERFALL_SINK_C_H
 
-#ifdef ENABLE_PYTHON
-#pragma push_macro("slots")
-#undef slots
-#include "Python.h"
-#pragma pop_macro("slots")
-#endif
-
 #include <gnuradio/fft/window.h>
 #include <gnuradio/qtgui/api.h>
 #include <gnuradio/sync_block.h>
diff --git a/gr-qtgui/include/gnuradio/qtgui/waterfall_sink_f.h b/gr-qtgui/include/gnuradio/qtgui/waterfall_sink_f.h
index bfd5fdc84..d08ae972f 100644
--- a/gr-qtgui/include/gnuradio/qtgui/waterfall_sink_f.h
+++ b/gr-qtgui/include/gnuradio/qtgui/waterfall_sink_f.h
@@ -11,13 +11,6 @@
 #ifndef INCLUDED_QTGUI_WATERFALL_SINK_F_H
 #define INCLUDED_QTGUI_WATERFALL_SINK_F_H
 
-#ifdef ENABLE_PYTHON
-#pragma push_macro("slots")
-#undef slots
-#include "Python.h"
-#pragma pop_macro("slots")
-#endif
-
 #include <gnuradio/fft/window.h>
 #include <gnuradio/qtgui/api.h>
 #include <gnuradio/sync_block.h>
diff --git a/gr-qtgui/lib/CMakeLists.txt b/gr-qtgui/lib/CMakeLists.txt
index fdb8f9f7b..e75eff0d7 100644
--- a/gr-qtgui/lib/CMakeLists.txt
+++ b/gr-qtgui/lib/CMakeLists.txt
@@ -76,18 +76,6 @@ if(ENABLE_COMMON_PCH)
     target_link_libraries(gnuradio-qtgui PRIVATE common-precompiled-headers)
 endif()
 
-include(GrPython)
-if(ENABLE_PYTHON)
-    target_compile_definitions(gnuradio-qtgui PUBLIC -DENABLE_PYTHON)
-    # INTERFACE/PRIVATE split so users of the library can choose how to link to Python
-    # (importantly, extension modules can avoid linking against Python and resolve
-    #  their own Python symbols at runtime through the Python interpreter's linking)
-    target_link_libraries(
-        gnuradio-qtgui
-        INTERFACE Python::Module
-        PRIVATE Python::Python)
-endif(ENABLE_PYTHON)
-
 set(qtgui_mod_includedir ${CMAKE_CURRENT_SOURCE_DIR}/../include/gnuradio/qtgui)
 qt5_wrap_cpp(
     qtgui_moc_sources
diff --git a/gr-qtgui/python/qtgui/bindings/sink_c_python.cc b/gr-qtgui/python/qtgui/bindings/sink_c_python.cc
index 05f3f6123..b7741a6d1 100644
--- a/gr-qtgui/python/qtgui/bindings/sink_c_python.cc
+++ b/gr-qtgui/python/qtgui/bindings/sink_c_python.cc
@@ -14,7 +14,7 @@
 /* BINDTOOL_GEN_AUTOMATIC(1)                                                       */
 /* BINDTOOL_USE_PYGCCXML(1)                                                        */
 /* BINDTOOL_HEADER_FILE(sink_c.h)                                        */
-/* BINDTOOL_HEADER_FILE_HASH(1bd021837fbd48564d6bdae9c445f527)                     */
+/* BINDTOOL_HEADER_FILE_HASH(be3c6d3500abbebbfdc610cb7a951715)                     */
 /***********************************************************************************/
 
 #include <pybind11/complex.h>
-- 
2.37.3

